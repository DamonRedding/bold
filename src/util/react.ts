import { MutableRefObject, Ref } from 'react'

export function composeRefs<T>(...refs: Array<Ref<T>>) {
    return (value: T) => refs.forEach(ref => setRef(ref, value))
}

export function setRef<T>(ref: Ref<T>, value: any) {
    if (typeof ref === 'function') {
        ref(value)
    } else if (ref) {
        (ref as MutableRefObject<T>).current = value
    }
}

/**
 * Returns the next sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getNextSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.nextElementSibling) {
        current = current.nextElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}

/**
 * Returns the previous sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getPreviousSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.previousElementSibling) {
        current = current.previousElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}
