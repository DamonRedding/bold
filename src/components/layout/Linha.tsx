import * as React from 'react'
import * as classnames from 'classnames'
import { UtilProps } from '../../util/Util'

export interface LinhaProps extends UtilProps {
    className?: string
    size?: 'medium'
}

export class Linha extends React.Component<LinhaProps, any> {

    render() {
        const classes = classnames({
            'medium': this.props.size && this.props.size === 'medium'
        })
        return (
            <hr className={classes} />
        )
    }

}
