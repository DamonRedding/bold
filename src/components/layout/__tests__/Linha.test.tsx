import * as React from 'react'
import { shallowRenderAndMatch } from '../../../__tests__/test.defaults'
import { Linha } from '../Linha'

describe('Linha', () => {
    it('render', () => {
        shallowRenderAndMatch(<Linha />)
    })
    it('render com tamanho mÃ©dio', () => {
        shallowRenderAndMatch(<Linha size='medium' />)
    })
})
