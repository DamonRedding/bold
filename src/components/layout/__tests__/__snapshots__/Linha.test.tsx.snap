// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Linha render 1`] = `
<hr
  className=""
/>
`;

exports[`Linha render com tamanho m√©dio 1`] = `
<hr
  className="medium"
/>
`;
